wolfcrypt/src/sha512.c

```diff
@@ -1615,7 +1615,7 @@ int wc_Sha512GetFlags(wc_Sha512* sha512, word32* flags)
 
 #if !defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)
 
-#if !defined(WOLFSSL_NOSHA512_224)
+#if !defined(WOLFSSL_NOSHA512_224) && !defined(HAVE_FIPS)
```

Comment:
This should be the construction we settled on earlier,
```
#if !defined(WOLFSSL_NOSHA512_224) && (!defined(HAVE_FIPS) || FIPS_VERSION_GE(5, 2)) && !defined(HAVE_SELFTEST)
```
and similar throughout.  As you saw, with FIPS 140-3 the algorithm macros are properly set up by the `configure` logic.  And for `fips-dev` purposes, we do want to be able to test them in a FIPS build.